alacritty_terminal/src/term/mod.rs

```diff
@@ -1839,15 +1839,15 @@ impl<T: EventListener> Handler for Term<T> {
         let num_cells = (cols - end).0;
 
         let line = cursor.point.line;
-        let line = &mut self.grid[line];
+        let line = &mut self.grid[line][..];
 
         for offset in 0..num_cells {
-            line[start + offset] = line[end + offset].clone();
+            line.swap(start.0 + offset, end.0 + offset);
         }
 
         // Clear last `count` cells in the row. If deleting 1 char, need to delete
         // 1 cell.
-        let end = cols - count;
+        let end = (cols - count).0;
         for cell in &mut line[end..] {
             *cell = bg.into();
         }
```

Comment:
These could probably also be combined maybe?

Reply:
Only if you start dynamically checking in a loop for boundaries, since those loops iterate for difference amount of cell.